### ISSUE TITLE
Difference between existing source plugins?

### ISSUE THREAD

**Issue by ansh** [butlerx/gatsby-source-google-spreadsheets#7]
I want to start off by saying that I am just a potential user who wants to use Google Sheets to source my data. The question I have, and I am sure many people who are in the same boat as I will have, is which is the best way to go about doing this?

I just want to understand where does this plugin come in? It seems like it is the most up-to-date method, but I want to make sure of my findings for future developers.

1. **Gatsby Docs recommended method:** https://www.gatsbyjs.org/blog/2019-07-23-google-sheets-gatsby-acroyoga-video-explorer/ is the blog post that the Gatsby Documentation recommends looking at when wanting to source data from Google Sheets. That uses the Node.JS API and makes all the data fetching happen through `gatsby-node.js` at build time. Is this method better than the route of going down plugins? What do you think?

2. **Using a plugin:** Currently on the [Gatbsy plugin list](https://www.gatsbyjs.org/plugins/) there are 2 different source plugins for Google sheets that are more popular than this one. What is the benefit of using this plugin over them besides the fact that it seems like the last commit was much more recent for this plugin than the others?
    - [gatsby-source-google-sheets](https://www.gatsbyjs.org/packages/gatsby-source-google-sheets/?=google%20sheets) with 1.9K downloads
    - [gatsby-source-google-spreadsheet](https://www.gatsbyjs.org/packages/gatsby-source-google-spreadsheet/?=google%20sheets) with 423 downloads
    - [gatsby-source-google-spreadsheets](https://www.gatsbyjs.org/packages/gatsby-source-google-spreadsheets/?=google%20sheets) (this one) with 360 downloads

**Comment by imyjimmy** [comment]
@ansh Good point. A lot of these plugins simply don't work, and I don't have the time to go through each one to figure out exactly which ones don't. The Google API has also changed a lot to the point where a lot of these tutorials are outdated.

**Comment by butlerx** [comment]
So question 2 first.
I created this project because the data wasn't in the right format for me from the other plugins. I wished to be able to export each sheet names as well as an unnamed version of each sheet All google APIs are the current versions and I run a build with the plugin once a day to verify it continues to work.

For question 1. That is an entirely valid route to go and gives you more control, the idea of this plugin is to abstract that away though as I used google sheets in multiple projects i just wanted to simplify my life with a shared plugin rather than a `gatsby-node.js` file
